### PR TITLE
Refactor functions and fix bugs

### PR DESCRIPTION
**Pull request by vbkaisetsu**
Remove some headers and use functions defined in the core library.

Cython does not support calling multiple template function (line `T func<vector<T>> (...)`) so `function_template_wrapper.h` can not be removed.

### CODE REVIEW

File: primitiv/_function.pyx [hunk 1]
```diff
@@ -17,12 +17,13 @@ class functions:
 
     @staticmethod
     def raw_input(shape, vector[float] data, Device device = None, Graph g = None):
-        if device is None:
-            device = Device.get_default()
+        cdef CppDevice *dev_p = NULL
```
Comment (odashi) [inline primitiv/_function.pyx]: Maybe it is better to implement a utility function to obtain correct values:
```python
cdef get_cpp_device(Device dev):
  return dev.wrapped if dev is not None else nullptr

cdef get_cpp_graph(Graph g):
  return g.wrapped if g is not None else nullptr
```
Then we can implement this function using just one statement:
```python
return wrapNode(
    func_input_node(
      normShape(shape).wrapped, data, get_cpp_device(dev), get_cpp_graph(g)))
```

**Comment by vbkaisetsu** [comment]
Fixed.

File: primitiv/_model.pyx [hunk 2]
```diff
@@ -39,9 +39,10 @@ cdef class Model:
         :type device: bool or None
 
         """
-        if device is None:
-            device = Device.get_default()
-        self.wrapped.load(pystr_to_cppstr(path), with_stats, device.wrapped[0])
+        cdef CppDevice *dev_p = NULL
```
Comment (odashi) [inline primitiv/_model.pyx]: Use `get_cpp_device()`.

File: primitiv/_parameter.pyx [hunk 3]
```diff
@@ -81,9 +81,11 @@ cdef class Parameter:
         :type device: primitiv.Device or None
 
         """
-        if device is None:
-            device = Device.get_default()
-        self.wrapped.init(normShape(shape).wrapped, initializer.wrapped[0], device.wrapped[0])
+        cdef CppDevice *dev_p = NULL
```
Comment (odashi) [inline primitiv/_parameter.pyx]: ditto

File: primitiv/_parameter.pyx [hunk 4]
```diff
@@ -97,9 +99,10 @@ cdef class Parameter:
         :type device: primitiv.Device or None
 
         """
-        if device is None:
-            device = Device.get_default()
-        self.wrapped.load(pystr_to_cppstr(path), with_stats, device.wrapped[0])
+        cdef CppDevice *dev_p = NULL
```
Comment (odashi) [inline primitiv/_parameter.pyx]: ditto

**Comment by vbkaisetsu** [comment]
Fixed.
Additionaly, `function_template_wrapper.h` was completely removed.

**Pull request merged**